mkt/tags/models.py

```diff
@@ -57,3 +56,23 @@ class AddonTag(ModelBase):
 
     class Meta:
         db_table = 'users_tags_addons'
+
+
+def attach_tags(objs, m2m_name=None):
+    """
+    Fetch tags from `objs` in one query and then attach them to a property on
+    each instance. The name of the property will be `m2m_name` + '_list'.
+
+    Assumes every instance in `objs` uses the same model, and needs `m2m_name`
+    to be set to the name of the m2m field between the model and Tag.
+    """
+    if m2m_name is None:
```

Comment:
This code won't catch the `attach_tags(webapps, 'tags')` call so I don't really see the point. It just makes me think `m2m_name` is going to default so something common like `'tags'` by looking at the signature. Is it worth adding an `arity` decorator?

```python
def arity(n):
    def wrapper(fn):
        def inner(*args, **kwargs):
            if len(args) != n:
                raise ValueError(
                    '{fn} takes exactly {n} positional arguments'.format(
                        fn=fn.__name__, n=n))
            else:
                return fn(*args, **kwargs)
        return inner
    return wrapper

@arity(2)
def foo(arg1, arg2, kwarg1, kwarg2):
    return arg1, arg2, kwarg1, kwarg2
```
```bash
>>> from arity import foo
>>> print(foo(1, 2, kwarg1=3, kwarg2=4))
(1, 2, 3, 4)
>>> print(foo(1, 2, 3, 4))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "arity.py", line 7, in inner
    fn=fn.__name__, n=n))
ValueError: foo takes exactly 2 positional arguments
>>> foo(1, 2, kwarg1=3, kwarg2=4, kwarg3=5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "arity.py", line 9, in inner
    return fn(*args, **kwargs)
TypeError: foo() got an unexpected keyword argument 'kwarg3'
```

This won't handle a `*args` case but if we just want to restrict to one position arg it does the trick.